af/shovel/centrifugation.py

```diff
@@ -771,8 +771,16 @@ def calc_measurement_flags(pgconn, flags_tbl, msm_tbl):
         ''')
         c.execute('''
         INSERT INTO {flags} SELECT msm_no, bool_or(anomaly), bool_or(confirmed) FROM (
-            SELECT msm_no, true AS anomaly, true AS confirmed FROM http_request_fp
-            WHERE msm_no IN (SELECT msm_no FROM {msm}) AND msm_no >= %s AND msm_no <= %s
+            SELECT http_request_fp.msm_no AS msm_no,
+                true AS anomaly,
+                true AS confirmed
+            FROM http_request_fp
+            JOIN fingerprint USING (fingerprint_no)
+            JOIN measurement_meta USING (msm_no)
+            JOIN report_blob USING (report_no)
+            WHERE origin_cc = probe_cc
+            AND http_request_fp.msm_no IN (SELECT msm_no FROM {msm})
+            AND http_request_fp.msm_no >= %s AND http_request_fp.msm_no <= %s
```

Comment:
If I understand SQL correcly, you can replace `http_request_fp.msm_no` with bare `msm_no` in this case.
I'm unsure if it improves readability or worsens it. :)